auction/src/lib.rs

```diff
@@ -358,6 +360,11 @@ pub mod pallet {
 				.ok_or(Error::<T>::MarketplaceNotFound)?;
 
 			marketplace.allowed_to_list(&who).ok_or(Error::<T>::AccountNotAllowedToList)?;
+			if let Some(collection_id) = nft.collection_id {
+				marketplace
+					.collection_allowed(&collection_id)
```

Comment:
Double check this but I got the feeling that we it's either you are on the account allow list or on the collection allow list.

Reply:
Hey @markopoloparadox 

Right now, I think we have this :
| Account | Collection | Account list collection
|--|--|--|
| Authorized | Authorized | Yes |
| Authorized | Banned| No |
| Banned| Authorized | No |
| Banned| Banned| No |

And if I understand, you want this :
| Account | Collection | Account list collection
|--|--|--|
| Authorized | Authorized | Yes |
| Authorized | Banned| Yes |
| Banned| Authorized | Yes |
| Banned| Banned| No |

The thing is, if I want to ban a user, I don't want him to list a collection even if the collection is whitelisted.
I understand that if I whitelist a collection, i want everyone to be able to list the collection.

Would the solution be to separate the public logic from the private logic ?